src/backend/optimizer/plan/createplan.c

```diff
@@ -1146,7 +1146,7 @@ static char** create_pxf_plan(char **segdb_file_map, RelOptInfo *rel, int total_
 	Relation relation = RelationIdGetRelation(planner_rt_fetch(scan_relid, ctx->root)->relid);
 	segdb_work_map = map_hddata_2gp_segments(uri_str, 
 											 total_segs, segs_participating,
-											 relation, NULL);
+											 relation, ctx->root->parse->jointree->quals);
```

Comment:
Can you ensure that the filters are only passed when the guc `pxf_enable_filter_pushdown` is enabled.  Such as is done here: https://github.com/apache/incubator-hawq/blob/316c5048f303513c40e8194b15024df12a3d76ad/src/backend/access/external/pxffilters.c#L554 

Currently if this guc is disabled filters should not be passed to PXF